### PR TITLE
Feature/speedup automaton

### PR DESCRIPTION
**Pull request by gnusi**

### CODE REVIEW

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/iresearch_s-pr/176/

**Comment by gnusi** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/iresearch_s-pr/179/

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/iresearch_s-pr/180/

File: core/utils/automaton.hpp [hunk 1]
```diff
@@ -129,49 +137,66 @@ class BooleanWeight {
...
     strm << '[' << l.min << ".." << l.max << ']';
     return strm;
   }
 
-  ValueType min{kNoLabel};
-  ValueType max{kNoLabel};
+  union {
+    int64_t ilabel;
+    struct {
+      uint32_t max;
```
Comment (Dronplane) [inline core/utils/automaton.hpp:180]: We should swap min and max members for Big-Endian

**Review by Dronplane — CHANGES_REQUESTED** [review]

File: tests/utils/levenshtein_utils_test.cpp [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #include "utils/levenshtein_utils.hpp"
 #include "utils/fstext/fst_table_matcher.hpp"
 #include "utils/utf8_utils.hpp"
+#include "draw-impl.h"
```
Comment (Dronplane) [inline tests/utils/levenshtein_utils_test.cpp:31]: Do we really need this?

**Review by Dronplane — APPROVED** [review]
LGTM

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/iresearch_s-pr/181/

**Pull request merged**